### PR TITLE
Add rescan

### PR DESCRIPTION
**Pull request by Rizato**
Any files that have a BlackboardArtifact already are eligible to be
rescanned.

This means, anything that has been scanned, or has had a hash lookup can
be rescanned.
Of course, these scans/hash lookups have to be in the same case for the
rescan option to show up.

### CODE REVIEW

**Comment by Rizato** [comment]
Full list of changes

*Switch SwarmIt to PolySwarm
* Add dialog for first scan to inform user they send the file to us
* Check size of file before upload
* Add First seen date to results
* Add last scanned date to results
* Add rescan option to files with previous results
* Add instructions to get api-key in autopsy

File: .gitignore [hunk 1]
```diff
@@ -21,7 +21,10 @@
 
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
-/swarmit/nbproject/private/
+/polyswarm/nbproject/private/
+/polyswarm/nbproject/private/
+/polyswarm/build/
+/swarmit/release/modules
```
Comment (mrtizmoatwork) [inline .gitignore:27]: shouldn't need these /swarmit/ dirs anymore

File: polyswarm/src/io/polyswarm/app/PolySwarmModule.java [hunk 2]
```diff
@@ -1,130 +1,131 @@
...
+                        }
+                        controller = null;
+
+                        Case newCase = (Case) evt.getNewValue();
+                        if (newCase != null) {
+                            // a new case has been opened: connect db, start worker thread
+                            try {
+                                controller = new PolySwarmController(newCase);
+                            } catch (TskCoreException ex) {
+                                LOGGER.log(Level.SEVERE, "Error changing case in ImageGallery.", ex);
```
Comment (mrtizmoatwork) [inline polyswarm/src/io/polyswarm/app/PolySwarmModule.java:117]: is "ImageGallery" correct here?

File: polyswarm/src/io/polyswarm/app/contextmenu/ScanAction.java [hunk 3]
```diff
@@ -1,86 +1,123 @@
...
+        super(menuItemStr);
+        this.abstractFile = abstractFile;
+    }
+
+    @Override
+    @org.openide.util.NbBundle.Messages({"ScanAction.dbError.message=Failed to record submission in pending submissions database.",
+        "ScanAction.submitError.message=Failed to submit file to PolySwarm.",
+        "ScanAction.messageDialog.title=PolySwarm Scan",
+        "ScanAction.sendFiles.message=Scanning files requires sending the file to PolySwarm.\nDo you want to continue?",
+        "ScanAction.tooLarge.message=File to large to submit to PolySwarm",
```
Comment (mrtizmoatwork) [inline polyswarm/src/io/polyswarm/app/contextmenu/ScanAction.java:56]: s/to/too

File: polyswarm/src/io/polyswarm/app/tasks/PendingTask.java [hunk 4]
```diff
@@ -106,38 +105,37 @@ public static void updateBlackboard(Case autopsyCase, Long abstractFileId, Artif
...
-        SwarmItController.createCustomArtifactAttribute(autopsyCase, attributeName, BlackboardAttribute.TSK_BLACKBOARD_ATTRIBUTE_VALUE_TYPE.STRING, assertion.name);
+        String attributeName = String.format(PolySwarmController.POLYSWARM_ARTIFACT_ATTRIBUTE_ASSERTION_NAME_FORMAT, assertion.name.toUpperCase());
+        PolySwarmController.createCustomArtifactAttribute(autopsyCase, attributeName, BlackboardAttribute.TSK_BLACKBOARD_ATTRIBUTE_VALUE_TYPE.STRING, assertion.name);
         addArtifactAttribute(autopsyCase, artifact, attributeName, assertion.getHumanReadableVerdict());
     }
 
     /**
-     * Creates a new BlackboardArtifact and adds the the AbstractFile
-     * We create a new one each task so the results are easy to read & easy to associate
+     * Creates a new BlackboardArtifact and adds the the AbstractFile We create a new one each task so the results are
```
Comment (mrtizmoatwork) [inline polyswarm/src/io/polyswarm/app/tasks/PendingTask.java:137]: ^ confusing sentence

**Pull request merged**